File: java/app/src/main/java/net/fbvictorhugo/j/barreirasanitaria/data/dao/BarreiraSanitariaDAO.java
package net.fbvictorhugo.j.barreirasanitaria.data.dao;

import net.fbvictorhugo.j.barreirasanitaria.data.model.BarreiraSanitaria;

import java.util.List;

public interface BarreiraSanitariaDAO {

    List<BarreiraSanitaria> listar();

    void inserir(BarreiraSanitaria barreiraSanitaria) throws Exception;

    void atualizar(BarreiraSanitaria barreiraSanitaria) throws Exception;

    BarreiraSanitaria procurarBarreira(long barreirId);

    String procurarNomeBarreira(long barreiraId);

}

File: kotlin/app/src/main/java/net/fbvictorhugo/k/barreirasanitaria/data/dao/BarreiraSanitariaDAO.kt
package net.fbvictorhugo.k.barreirasanitaria.data.dao

import net.fbvictorhugo.k.barreirasanitaria.data.model.BarreiraSanitaria

interface BarreiraSanitariaDAO {

    fun listar(): ArrayList<BarreiraSanitaria>

    @Throws(Exception::class)
    fun inserir(barreiraSanitaria: BarreiraSanitaria)

    @Throws(Exception::class)
    fun atualizar(barreiraSanitaria: BarreiraSanitaria)

    fun procurarBarreira(barreiraId: Long): BarreiraSanitaria?

    fun procurarNomeBarreira(barreiraId: Long): String

}
